air/src/trace/rows.rs

```diff
@@ -7,7 +7,7 @@ use vm_core::Felt;
 
 /// Represents the types of errors that can occur when converting from and into [`RowIndex`] and
 /// using its operations.
-#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+#[derive(Debug, thiserror::Error)]
 pub enum RowIndexError<T> {
```

Comment:
Not from this PR, but I think we should remove that generic `T` and replace with `u32`. The "trace cannot be greater than 2^32" assumption is here to stay (at the very least until we decide to use a field over than Goldilocks), so I think it's better to just use `u32`.

Reply:
Addressed all of your comments now, thank you!

Because we use various integer types (i32, usize, u64) with this error we can't use a specific one in the `InvalidValue` variant. So instead of using a generic parameter I'm using a formatted string instead now which includes the type so we produce nice error messages.